src/Microsoft.AspNet.Razor/Runtime/TagHelpers/TagStructure.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// The structure the tag should be written in.
+    /// </summary>
+    public enum TagStructure
+    {
+        /// <summary>
+        /// If no other tag helper applies to the same element and specifies a <see cref="TagStructure"/>,
+        /// <see cref="NormalOrSelfClosing"/> will be used.
+        /// </summary>
+        Unspecified,
+
+        /// <summary>
+        /// Tag can be written as &lt;my-tag-helper&gt;&lt;/my-tag-helper&gt; or &lt;my-tag-helper /&gt;.
+        /// </summary>
+        NormalOrSelfClosing,
+
+        /// <summary>
+        /// Tag can be written as &lt;my-tag-helper&gt; or &lt;my-tag-helper /&gt;.
+        /// </summary>
+        /// <remarks>Tags with a <see cref="WithoutEndTag"/> structure will never have any content.</remarks>
```

Comment:
Try to stick to usual meanings of "tag" and "element". This structure, for example, requires a single tag for the entire element. So it's the _element_ that can be written these ways and that will always be empty.